src/model/FileParser.kt

```diff
@@ -0,0 +1,93 @@
+import java.io.File
+
+/**
+ * Defines a FileParser object, containing different operations on the content of a given .csv file
+ *
+ * created by AlexandraV on 12/25/2018
+ */
+data class FileParser(var fileName: String, var userCoordinate: Coordinate) {
+    val coffeeShopsList: MutableList<CoffeeShop> = mutableListOf()
+    private val lines: List<String>? = File(fileName).useLines { it.toList() } // the reader is closed automatically
+
+    /**
+     *  parses the lines of the .csv file and converts them into CoffeeShops objects, if possible
+     *  All the lines that cannot be converted to CoffeeShops objects will be written into  invalid_coffee_shops.txt file
+     */
+    fun parseCoffeeShopsList(n: Int) {
+        val invalidCoffeeShopsList: MutableList<String> = mutableListOf()
+        if (lines != null) {
+            var lineNumber = 1
+            lines.forEach { it ->
+                val result: List<String> = it.split(",").map { it.trim() }
+                if (isCSVLineValid(result)) {
+                    val shopCoordinate = Coordinate(result[1], result[2])
+                    insertShopIntoList(CoffeeShop(result[0], shopCoordinate.distanceTo(userCoordinate.xCoordinate, userCoordinate.yCoordinate)), n)
+                } else {
+                    invalidCoffeeShopsList.add("Line $lineNumber: $it \n")
+                }
+                lineNumber++
+            }
+
+            if (n <= coffeeShopsList.size) {
+                coffeeShopsList.subList(0, n).forEach {
+                    println("${it.shopName}, ${it.shopDistanceToUser}")
```

Comment:
What is the problem you are trying to solve by having `println`s all over the place?